packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/usePaymentInformation.js

```diff
@@ -115,6 +136,27 @@ export const usePaymentInformation = props => {
         onCompleted: onPaymentDetailsQueryCompleted
     });
 
+    const handleExiredPaymentError = useCallback(() => {
+        clearPaymentDetails({ variables: { cartId } });
+        resetReviewOrderButtonClicked();
+        setHasData(false);
+        onError();
```

Comment:
Use a try-catch in the checkout page so you don't have to thread the `onError` down to the child nor do you need to reset the review order button.

Reply:
If I have the try-catch and the throw statement in the `useCheckoutPage` talon, how will the children catch it? It will only propagate upwards, right?

Reply:
I commented this in multiple places, but for posterity: `placeOrderError` can be passed down (or some combination of the error).

Reply:
You are right, I can have the try-catch and send the rest of the props down the `PaymentInformation` component/talon. But I feel, sending down a function to call when an error occurs is better than having some logic in try-catch in the parent and some in the child even though the function sent down to the child executes in the parent context.

That said, the function in question, `revertPaymentInformationDone` sent as `onError` is called as the last function in `handleExiredPaymentError`. If instead of sending it as a prop, we call it in the try-catch, it will be called first. The child component won't have control over when to call the function.

TLDR, I feel comfortable sending it as a prop and having the child components handle the error and the onError handler. In this case, it is `PaymentInformation`, but once we expand this to check for errors in `ShippingInformation` and `ShippingMethods` they can call the handlers themselves when they think is the right time.

Reply:
OK I get. it! In the future we may get other errors during `placeOrder` that are related to other checkout components besides payment info. So this pattern would delegate the responsibility for handling the error to the component that caused it. 

I'm on board with this but I do think we can at least move `resetReviewOrderButtonClicked` to `useCheckoutPage` handler as that's a global concern we will want to do every time we get an error.